### PR TITLE
FIX unset categorie

### PR DESCRIPTION
**Pull request by atm-quentin**
# Fix Unset category
If a user hasn't rights to modify category, it'll empty categories

### CODE REVIEW

File: htdocs/societe/card.php [hunk 1]
```diff
@@ -519,15 +519,15 @@
...
-
-					// Customer categories association
-					$custcats = GETPOST( 'custcats', 'array' );
-					$object->setCategories($custcats, 'customer');
-
-					// Supplier categories association
-					$suppcats = GETPOST('suppcats', 'array');
-					$object->setCategories($suppcats, 'supplier');
-
+					if(!empty($user->rights->categorie->creer)){
```
Comment (eldy) [inline htdocs/societe/card.php]: The permission categorie->creer is permission to add/delete a category among other categories). Right to admin the tags.
Here, we edit a category as a property of a thirdparty, so permission must be $user->rights->societe->creer and seems already tested before.

Comment (atm-quentin) [reply htdocs/societe/card.php]: Ok, my bad.
Here's my problem then : If a user hasn't right to read categorie, and update a thirdparty, it'll empty thirdparty's categories. So do you prefer to check $user->rights->categorie->lire ?

Comment (eldy) [reply htdocs/societe/card.php]: I see.
Yes, just disable update code with test on "$user->rights->categorie->lire "
Please add a comment to explain this case too so we understand why we have a test on read to update.

**Comment by atm-quentin** [comment]
fixed creer into lire

**Pull request merged**